administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -289,70 +289,86 @@ public static function chmod($path, $mode, $is_dir = false)
+   * @param   boolean $anim                   true=preserve animation if any
    * @return  boolean True on success, false otherwise
    * @since   1.0.0
    */
-  public static function resizeImage(&$debugoutput, $src_file, $dest_file, $useforresizedirection,
-                                     $new_width, $thumbheight, $method, $dest_qual, $max_width = false, $cropposition, $angle = 0)
+  public static function resizeImage(&$debugoutput, $src_file, $dest_file, $settings,
+                                     $new_width, $new_height, $method, $dest_qual, $cropposition = false, $angle = 0, $metadata = false, $anim = false)
   {
+
+    // animated gifs: https://phpimageworkshop.com/tutorial/5/manage-animated-gif-with-imageworkshop.html
+
     $config = JoomConfig::getInstance();
 
-    // Ensure that the pathes are valid and clean
+    // Ensure that the paths are valid and clean
     $src_file  = JPath::clean($src_file);
     $dest_file = JPath::clean($dest_file);
 
-    // Doing resize instead of thumbnail, copy original and remove it.
-    $imginfo = getimagesize($src_file);
-
-    if(!$imginfo)
+    // Analysis of the source image, if image is valid
+    //$imginfo = getimagesize($src_file, $src_metainfo);
+    if(!($imginfo = JoomFile::analyseImage($src_file)))
     {
       $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_FILE_NOT_FOUND').'<br />';
-
       return false;
     }
 
-    // GD can only handle JPG & PNG images
-    if(    $imginfo[2] != IMAGETYPE_JPEG
-       &&  $imginfo[2] != IMAGETYPE_PNG
-       &&  $imginfo[2] != IMAGETYPE_GIF
+    // GD can only handle JPG, PNG and GIF images
+    if(    $imginfo['type'] != 'JPG'
+       &&  $imginfo['type'] != 'PNG'
+       &&  $imginfo['type'] != 'GIF'
        &&  ($method == 'gd1' || $method == 'gd2')
       )
     {
       $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_GD_ONLY_JPG_PNG').'<br />';
-
       return false;
     }
+    $src_imagetype = $imginfo['type'];
 
-    $imagetype = array(1 => 'GIF', 2 => 'JPG', 3 => 'PNG', 4 => 'SWF', 5 => 'PSD',
-                       6 => 'BMP', 7 => 'TIFF', 8 => 'TIFF', 9 => 'JPC', 10 => 'JP2',
-                       11 => 'JPX', 12 => 'JB2', 13 => 'SWC', 14 => 'IFF');
-
-    $imginfo[2] = $imagetype[$imginfo[2]];
+    // get the desired image type out of the destination path
+    $tmp = explode('.', $dest_file);
+    $dest_imgtype = strtolower(end($tmp));
+    if ($dest_imgtype == 'jpg' || $dest_imgtype == 'jpeg' || $dest_imgtype == 'jpe' || $dest_imgtype == 'jif' || $dest_imgtype == 'jfif' || $dest_imgtype == 'jfi')
+    {
+      $dest_imgtype = 'JPG';
+    }
+    elseif ($dest_imgtype == 'gif')
+    {
+      $dest_imgtype = 'GIF';
+    }
+    elseif ($dest_imgtype == 'png')
+    {
+      $dest_imgtype = 'PNG';
+    }
+    else
+    {
+      $dest_imgtype = 'UNKNOWN';
+      $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_GD_ONLY_JPG_PNG').'<br />';
+      return false;
+    }
```

Comment:
detect the desired file type for the resized image out of the file-ending of the path for the destitation image